### PR TITLE
feat: Perform deletion instead of mark as deleted

### PR DESCRIPTION
**Pull request by yevhen-onykiienko-snkeos**
Issue #, if available:

Description of changes:
calling `deleteResource` deletes a resource from the database instead of marking it as deleted

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mario-degenhardt-snkeos — DISMISSED** [review]

**Review by christopher-watanabe-snkeos — CHANGES_REQUESTED** [review]
I would like to see cdh-tests passing with these changes before we merge, since other deployments (intx) will be updated when this is merged to snkeos-main.

**Review by christopher-watanabe-snkeos — COMMENTED** [review]
Looks like the tests run fine on main. I hope that the changes to the GKV generation to the cdh-tests solve the problem of concurrency. 

In the meantime, do you know if the tests check whether a delete resource is actually deleted? If not @mario-degenhardt-snkeos might be adding something like that to his [right to be forgotten tests](https://github.com/snkeos/cdh-tests/tree/feat/right-to-be-forgotten-patient-mgnt)

**Comment by mario-degenhardt-snkeos** [comment]
@christopher-watanabe-snkeos in cdh-tests the function that checks if there is data available or not is called get_item_from_resource_table. It provides the option to filter by status AVAILABLE or to return all

**Comment by christopher-watanabe-snkeos** [comment]
> @christopher-watanabe-snkeos in cdh-tests the function that checks if there is data available or not is called get_item_from_resource_table. It provides the option to filter by status AVAILABLE or to return all

I see. So we can adapt tests for this change to get all items regardless of status and ensure that no items are returned.

File: src/dataServices/dynamoDbDataService.ts
Comment (yevhen-onykiienko-snkeos) [inline src/dataServices/dynamoDbDataService.ts:161]: I should probably delete the commented code. or?

Comment (christopher-watanabe-snkeos) [reply src/dataServices/dynamoDbDataService.ts:161]: I would. If we ever need to rollback, we can revert the changes made in this commit.

File: src/dataServices/dynamoDbDataService.ts
Comment (christopher-watanabe-snkeos) [inline src/dataServices/dynamoDbDataService.ts:190]:
```suggestion
        const itemServiceResponse = await this.readResource({ resourceType, id, tenantId });
```

File: src/dataServices/dynamoDbDataService.ts
Comment (christopher-watanabe-snkeos) [inline src/dataServices/dynamoDbDataService.ts:192]:
```suggestion
        return this.deleteVersionedResource(id, parseInt(versionId, 10), tenantId);
```

**Review by christopher-watanabe-snkeos — COMMENTED** [review]
Only found a couple missing semi-colons. Otherwise, happy to see that the tests are passing.

**Review by christopher-watanabe-snkeos — APPROVED** [review]
Please wait for @mario-degenhardt-snkeos 's review before merging

**Review by mario-degenhardt-snkeos — APPROVED** [review]

**Pull request merged**
